common/achievement/util.ts

```diff
@@ -62,12 +63,12 @@ export async function getBucketContext(userID: string, relation?: string): Promi
 
     let subcourses = [];
     if (!relationType || relationType === 'subcourse') {
-        let subcourseWhere = whereClause;
-        if (userType === 'student') {
-            subcourseWhere = { ...subcourseWhere, subcourse_instructors_student: { some: { studentId: id } } };
-        } else {
-            subcourseWhere = { ...subcourseWhere, subcourse_participants_pupil: { some: { pupilId: id } } };
-        }
+        delete whereClause[`${userType}Id`];
+        const userClause =
+            userType === 'student'
+                ? { subcourse_instructors_student: { some: { studentId: id } } }
+                : { subcourse_participants_pupil: { some: { pupilId: id } } };
+        const subcourseWhere = { ...whereClause, ...userClause };
```

Comment:
I think the upper solution was merged in from another branch, since this was not yet implemented when branching of this one. I'd say though, that I like the new solution slightly better, what do you think?

Reply:
Makes sense 👍